app/renderer/actions/Inspector.js

```diff
@@ -125,27 +147,13 @@ export function selectElement (path) {
     }
     dispatch({type: SET_EXPANDED_PATHS, paths: expandedPaths});
 
+
     // Find the optimal selection strategy. If none found, fall back to XPath.
     const strategyMap = _.toPairs(getLocators(selectedElementAttributes, sourceXML));
     const [optimalStrategy, optimalSelector] = strategyMap.length > 0 ? strategyMap[strategyMap.length - 1] : ['xpath', selectedElementXPath];
 
-    // Get the information about the element
-    let {elementId, variableName, variableType} = await callClientMethod({
-      strategy: optimalStrategy,
-      selector: optimalSelector,
-    });
-
-    if (!elementId) {
-      // If the element is stale, unselect and reload the source
-      dispatch({type: UNSELECT_ELEMENT});
-      return await applyClientMethod({methodName: 'source'})(dispatch);
-    } 
-
-    // Set the elementId, variableName and variableType for the selected element 
-    // (check first that the selectedElementPath didn't change, to avoid race conditions)
-    if (getState().inspector.selectedElementPath === path) {
-      dispatch({type: SET_SELECTED_ELEMENT_ID, elementId, variableName, variableType});
-    }
+    // Debounce find element so that if another element is selected shortly after, cancel the previous search
+    findElement(optimalStrategy, optimalSelector, dispatch, getState, path);
```

Comment:
shouldn't we await for it?

Reply:
The 'debounced' function doesn't return a promise, it gets called 1000 ms after the most recent call to it.

Reply:
The documentation says 
> Subsequent calls to the debounced function return the result of the last func invocation.

https://lodash.com/docs/4.17.4#debounce

Reply:
Okay that makes sense then that if ALL of the calls to debounce return the promise then we can use await. It would be problem if the subsequent calls to debounce terminated and didn't return anything.